crates/sui-core/src/authority/authority_store.rs

```diff
@@ -62,52 +69,95 @@ pub struct SuiDataStore<const ALL_OBJ_VER: bool, const USE_LOCKS: bool, S> {
     /// composite key of the SuiAddress of their owner and the object ID of the object.
     /// This composite index allows an efficient iterator to list all objected currently owned
     /// by a specific user, and their object reference.
+    ///
+    /// Consistency requirements:
+    /// - Only exists to satisfy read queries from clients, and should be moved to IndexStore
+    /// - Eventual consistency with respect to certificate processing - the owner index must
+    ///   eventually be updated after a certificate is processed.
+    /// - Read-after-write consistency can be provided if required, such that if client has a
+    ///   signed effects cert they cannot be served the old owner_index entry.
     owner_index: DBMap<(SuiAddress, ObjectID), ObjectRef>,
 
     /// This is map between the transaction digest and transactions found in the `transaction_lock`.
     /// NOTE: after a lock is deleted (after a certificate is processed) the corresponding entry here
     /// could be deleted, but right now this is only done on gateways, not done on authorities.
+    ///
+    /// Consistency requirements:
+    /// - Must never contain an entry for a TransactionDigest unless locks have been succesfully
+    ///   acquired for that tx.
+    /// - Read-after-write consistency for the client: If the Validator::transaction RPC returns
+    ///   successfully, this map must be up to date. If the RPC fails before returning, the client
```

Comment:
Is it ok here if this map is eventually up to date? I think that is the case. the request to sign does return the signed transaction, and the locks must be updated before we return. However, it is not clear this table needs to be updated before we return, as long as we ensure that it is eventually updated (I think -- lets carefully check this).